### PR TITLE
Making sure the application's exit code is properly propagated back to the OS

### PR DESCRIPTION
**Pull request by hkaiser**
This fixes the issue that some tests don't report their errors back to CircleCI.

### CODE REVIEW

**Comment by hkaiser** [comment]
@NK-Nikunj please verify

File: src/hpx_wrap.cpp [hunk 1]
```diff
@@ -61,16 +61,17 @@ namespace hpx_start
...
         // call to the main() function
         int return_value = main(argc, argv);
-        #endif
+#endif
 
-        //Finalizing the HPX runtime
-        return hpx::finalize(return_value);
+        // Finalizing the HPX runtime
+        hpx::finalize();
+        return return_value;s
```
Comment (NK-Nikunj) [inline src/hpx_wrap.cpp]: @hkaiser there's an extra 's'

**Comment by NK-Nikunj** [comment]
@hkaiser Just ran the builds and checked. This seems to resolve the issue. You were right, `hpx::finalize` does not propagate the return value (I cross-validated that as well).

File: src/hpx_wrap.cpp (hunk 1 above)
Comment (hkaiser) [reply src/hpx_wrap.cpp]: That's fixed now, thanks!

File: src/hpx_wrap.cpp [hunk 2]
```diff
@@ -61,16 +61,17 @@ namespace hpx_start
...
         // call to the main() function
         int return_value = main(argc, argv);
-        #endif
+#endif
 
-        //Finalizing the HPX runtime
-        return hpx::finalize(return_value);
+        // Finalizing the HPX runtime
+        hpx::finalize();
+        return return_value;
```
Comment (biddisco) [inline src/hpx_wrap.cpp]: OK. I thought that finalize could return it, but if this works properly

Comment (hkaiser) [reply src/hpx_wrap.cpp]: Yah, I thought that as well … :/
But it works this way, we verified.

**Comment by hkaiser** [comment]
The fact that it works is confirmed by a couple of tests now failing. Those were failing before but were not reported as such.

File: src/hpx_wrap.cpp (hunk 2 above)
Comment (biddisco) [reply src/hpx_wrap.cpp]: would the "proper" fix to be to make sure that `return hpx::finalize(return_value);` did what we expect it to?

Comment (hkaiser) [reply src/hpx_wrap.cpp]: I think such an overload for `hpx::finalize()` would be possible but dangerous. We already have an overload taking a `double`.

**Pull request merged**